### PR TITLE
test(commons): migrate to vitest

### LINKED ISSUE #3059: Maintenance: migrate test runner to vitest for commons package
### Summary

To continue the migration from Jest to Vitest we need to move over the tests for the `commons` package to use the new test runner.

### Why is this needed?

So that we can continue our efforts of removing dev dependencies and streamline the tests.

### Which area does this relate to?

Commons

### Solution

_No response_

### Acknowledgment

- [X] This request meets [Powertools for AWS Lambda (TypeScript) Tenets](https://docs.powertools.aws.dev/lambda/typescript/latest/#tenets)
- [ ] Should this be considered in other Powertools for AWS Lambda languages? i.e. [Python](https://github.com/aws-powertools/powertools-lambda-python/), [Java](https://github.com/aws-powertools/powertools-lambda-java/), and [.NET](https://github.com/aws-powertools/powertools-lambda-dotnet/)

### Future readers

Please react with 👍 and your use case to help us understand customer demand.

### PR DESCRIPTION
**Pull request by dreamorosi**
### Changes

> Please provide a summary of what's being changed

This PR switches the test runner of the `commons` package from Jest to Vitest, continuing the migration effort for the project.

The PR is relatively small and focuses primarily in: 1/ changing the imports to `vitest` specific ones, 2/ standardizing the test headers, and 3/ updating the commands & configs to suit the new runner.

The only notable change of the PR is that [Vitest supports type testing](https://vitest.dev/guide/testing-types.html), which opens up new possibilities for our tests. Concretely this PR introduces the first "unit" test for types around the `LambdaInterface` type. We'll be experimenting with more tests of this kind in the future.

> Please add the issue number below, if no issue is present the PR might get blocked and not be reviewed

**Issue number:** closes #3059

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: packages/commons/package.json [hunk 1]
```diff
@@ -10,11 +10,11 @@
     "access": "public"
   },
   "scripts": {
-    "test": "npm run test:unit",
-    "test:unit": "jest --group=unit --detectOpenHandles --coverage --verbose",
-    "jest": "jest --detectOpenHandles --verbose",
+    "test": "vitest --run",
+    "test:unit": "vitest --run",
+    "test:unit:coverage": "vitest --run tests/unit --coverage.enabled --coverage.thresholds.100 --coverage.include='src/**'",
```
Comment (leandrodamascena) [inline packages/commons/package.json:15]: 100% of coverage threshold, you're doing a great job dude!

File: packages/commons/tests/unit/LRUCache.test.ts [hunk 2]
```diff
@@ -86,7 +82,7 @@ describe('Class: LRUMap', () => {
       expect(value).toBeUndefined();
     });
 
-    test('when called it marks the item as the most recently used', () => {
+    it('marks the item as the most recently used', () => {
```
Comment (leandrodamascena) [inline packages/commons/tests/unit/LRUCache.test.ts:85]: I'm not sure if this test is testing something that hasn't been tested before.
You may don't need this.

File: packages/commons/tests/unit/LRUCache.test.ts [hunk 3]
```diff
@@ -146,7 +142,7 @@ describe('Class: LRUMap', () => {
       expect(cache.get('a')).toBeUndefined();
     });
 
-    test('when called on an empty cache it does nothing', () => {
+    it('it does nothing when called on an empty cache', () => {
```
Comment (leandrodamascena) [inline packages/commons/tests/unit/LRUCache.test.ts:145]:
```suggestion
    it('it does nothing when called on an non-existing key', () => {
```

File: packages/commons/tests/unit/Utility.test.ts [hunk 4]
```diff
@@ -81,10 +77,10 @@ describe('Class: Utility', () => {
   });
 
   describe('Method: isColdStart', () => {
```
Comment (leandrodamascena) [inline packages/commons/tests/unit/Utility.test.ts:79]: Is this block repeated?

**Review by leandrodamascena — APPROVED** [review]
Hi @dreamorosi! I left some small comments, but it doesn't affect this migration if you don't accept it.

Approving.

File: packages/commons/tests/unit/LRUCache.test.ts (hunk 2 above)
Comment (dreamorosi) [reply packages/commons/tests/unit/LRUCache.test.ts:85]: You're right, good catch!

Removed

File: packages/commons/tests/unit/LRUCache.test.ts (hunk 3 above)
Comment (dreamorosi) [reply packages/commons/tests/unit/LRUCache.test.ts:145]: Good point!

File: packages/commons/tests/unit/Utility.test.ts (hunk 4 above)
Comment (dreamorosi) [reply packages/commons/tests/unit/Utility.test.ts:79]: It's very similar to getColdStart(), and in the future we could remove one of the two to be honest, but for now we need to test both for coverage.

**Review by leandrodamascena — APPROVED** [review]

File: packages/commons/tests/unit/LRUCache.test.ts (hunk 2 above)
Comment (dreamorosi) [reply packages/commons/tests/unit/LRUCache.test.ts:85]: Turns out the test was needed, if I remove it coverage drops and CI fails.

**Review by leandrodamascena — APPROVED** [review]

**Pull request merged**